crates/noirc_evaluator/src/ssa_refactor/ir/instruction.rs

```diff
@@ -92,6 +92,12 @@ pub(crate) enum Instruction {
     /// Performs a function call with a list of its arguments.
     Call { func: ValueId, arguments: Vec<ValueId> },
 
+    // TODO NEEDED?
+    // /// Performs an "oracle" - an external function call with a list of its arguments.
+    // /// These are generally unconstrained functions that provide some value lookup.
+    // /// They may result in constraints outside of the context known to Noir.
+    // ForeignCall { func: String, arguments: Vec<ValueId> },
```

Comment:
Do you have the todo needed because you think you can mock this using the `Call` instruction? I think we could technically fetch the func name using the func value id from the `foreign_functions` map in the DataFlowGraph, but that may be ugly

Reply:
Hmm I thought so, but it'd be good feedback if not